packages/yarnpkg-core/tests/semverUtils.test.ts

```diff
@@ -112,6 +112,8 @@ describe(`semverUtils`, () => {
       [[`<=1.5.3`, `1.5.3`], `1.5.3`],
       [[`1.5.3`, `1.5.3`], `1.5.3`],
       [[`1.5.0`, `1.5.3`], null],
+      [[`~1.0.1 || ~1.0.2`], `~1.0.1`],
+      [[`~1.0.1 || ~1.0.2`, `~1.0.1 || ~1.0.2`], `~1.0.1`],
```

Comment:
Can you add a test that crashes the process (for example perhaps something like:
```suggestion
      [[`~1.0.1 || ~1.0.2`, `~1.0.1 || ~1.0.2`], `~1.0.1`],
      [[`1.0.0${` || 1.0.0`.repeat(1000)}`], `~1.0.1`],
```

Reply:
Sure, I added this.